src/mui/input/ReferenceManyInput.js

```diff
@@ -0,0 +1,222 @@
+import React, {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import debounce from 'lodash.debounce';
+import Labeled from '../input/Labeled';
+import {
+    crudGetMany as crudGetManyAction,
+    crudGetMatching as crudGetMatchingAction
+} from '../../actions/dataActions';
+import {getPossibleReferences} from '../../reducer/references/possibleValues';
+
+const referenceSource = (resource, source) => `${resource}@${source}`;
+const noFilter = () => true;
+
+/**
+ * An Input component for choosing many reference records. Useful for 'hasMany' relationship.
+ *
+ * This component fetches the possible values in the reference resource
+ * (using the `CRUD_GET_MATCHING` REST method), then delegates rendering
+ * to a subcomponent, to which it passes the possible choices
+ * as the `choices` attribute.
+ *
+ * Use it with a selector component as child, like `<SelectManyInput>`.
+ *
+ * @example
+ * export const CommentEdit = (props) => (
+ *     <Edit {...props}>
+ *         <SimpleForm>
+ *             <ReferenceManyInput label="Post" source="post_id" reference="posts">
+ *                 <SelectManyInput type="object" optionText="title" />
+ *             </ReferenceManyInput>
+ *         </SimpleForm>
+ *     </Edit>
+ * );
+ *
+ * By default, restricts the possible values to 25. You can extend this limit
+ * by setting the `perPage` prop.
+ *
+ * @example
+ * <ReferenceManyInput
+ *      source="post_id"
+ *      reference="posts"
+ *      perPage={100}>
+ *     <SelectManyInput type="object" optionText="title" />
+ * </ReferenceManyInput>
+ *
+ * By default, orders the possible values by id desc. You can change this order
+ * by setting the `sort` prop (an object with `field` and `order` properties).
+ *
+ * @example
+ * <ReferenceManyInput
+ *      source="post_id"
+ *      reference="posts"
+ *      sort={{ field: 'title', order: 'ASC' }}>
+ *     <SelectManyInput type="object" optionText="title" />
+ * </ReferenceManyInput>
+ *
+ * Also, you can filter the query used to populate the possible values. Use the
+ * `filter` prop for that.
+ *
+ * @example
+ * <ReferenceManyInput
+ *      source="post_id"
+ *      reference="posts"
+ *      filter={{ is_published: true }}>
+ *     <SelectManyInput type="object" optionText="title" />
+ * </ReferenceManyInput>
+ *
+ * The enclosed component may filter results. ReferenceManyInput passes a `setFilter`
+ * function as prop to its child component. It uses the value to create a filter
+ * for the query - by default { q: [searchText] }. You can customize the mapping
+ * searchText => searchQuery by setting a custom `filterToQuery` function prop:
+ *
+ * @example
+ * <ReferenceManyInput
+ *      source="post_id"
+ *      reference="posts"
+ *      filterToQuery={searchText => ({ title: searchText })}>
+ *     <SelectManyInput type="object" optionText="title" />
+ * </ReferenceManyInput>
+ */
+export class ReferenceManyInput extends Component {
```

Comment:
my 2c: as there is a lot of common code with `ReferenceInput`, why don't you extend `ReferenceInput` instead of `Component`?